internal_public.go

```diff
@@ -36,15 +36,15 @@ import (
 )
 
 type (
-	// WorkflowHistoryIterator - To fetch history given a continuation token.
-	WorkflowHistoryIterator func(nextToken []byte) (*s.History, []byte, error)
+	// GetHistoryPage - Fetches a history page given a continuation token.
+	GetHistoryPage func(nextToken []byte) (*s.History, []byte, error)
```

Comment:
I believe GetHistoryPage is a better name as it is going to return the same page given the same token.